test/databases.test.js

```diff
@@ -2898,6 +2898,57 @@ const user = { renderingId: 1 };
           }
         });
       });
+
+      describe('request on the resources getter with restrictFieldsOnRootModel flag', () => {
+        it('should only retrieve requested fields with the flag', async () => {
+          expect.assertions(6);
+          const { models } = initializeSequelize();
+          const spy = jest.spyOn(scopeManager, 'getScopeForUser').mockReturnValue(null);
+          const params = {
+            ...baseParams,
+            fields: { address: 'id,line,zipCode' },
+            page: { number: '1' },
+            restrictFieldsOnRootModel: true,
+          };
+
+          try {
+            const result = await new ResourcesGetter(models.address, null, params).perform();
+            expect(result[0]).not.toHaveLength(0);
+            expect(result[0][0].dataValues).toHaveProperty('id');
+            expect(result[0][0].dataValues).toHaveProperty('line');
+            expect(result[0][0].dataValues).toHaveProperty('zipCode');
+            expect(result[0][0].dataValues).not.toHaveProperty('city');
+            expect(result[0][0].dataValues).not.toHaveProperty('country');
+          } finally {
+            spy.mockRestore();
+            connectionManager.closeConnection();
+          }
+        });
+
+        it('should ignore requested fields without the flag', async () => {
+          expect.assertions(6);
+          const { models } = initializeSequelize();
+          const spy = jest.spyOn(scopeManager, 'getScopeForUser').mockReturnValue(null);
+          const params = {
+            ...baseParams,
+            fields: { address: 'id,line,zipCode' },
+            page: { number: '1' },
+          };
+
+          try {
+            const result = await new ResourcesGetter(models.address, null, params).perform();
+            expect(result[0]).not.toHaveLength(0);
+            expect(result[0][0].dataValues).toHaveProperty('id');
+            expect(result[0][0].dataValues).toHaveProperty('line');
+            expect(result[0][0].dataValues).toHaveProperty('zipCode');
+            expect(result[0][0].dataValues).toHaveProperty('city');
+            expect(result[0][0].dataValues).toHaveProperty('country');
+          } finally {
+            spy.mockRestore();
+            connectionManager.closeConnection();
+          }
+        });
+      });
```

Comment:
i know this test seems not to make any sense, but let's prevent new regressions...